File: auth-server/src/main/java/example/demo/controllers/AuthController.java
package example.demo.controllers;

import auth.dto.AuthRequest;
import lombok.extern.slf4j.Slf4j;
import org.springframework.messaging.handler.annotation.MessageMapping;
import org.springframework.stereotype.Controller;
import reactor.core.publisher.Mono;

@Controller
@Slf4j
public class AuthController {

    @MessageMapping("auth-handler")
    public Mono<AuthRequest> auth(String user) {
        log.info("User log with this userName {}", user);
        return "user".equals(user) ? Mono.just(new AuthRequest(true)) : Mono.just(new AuthRequest(false));
    }
}

File: auth-server/src/main/kotlin/example/demo/controllers/AuthController.kt
package example.demo.controllers;

import auth.dto.AuthRequest
import auth.dto.AuthResponse
import auth.dto.CodeResponse
import example.demo.services.AuthenticationService
import mu.KotlinLogging
import org.springframework.messaging.handler.annotation.MessageMapping
import org.springframework.stereotype.Controller

@Controller
class AuthController(val authenticationService: AuthenticationService) {
    private val logger = KotlinLogging.logger {}

    @MessageMapping("get-auth-code")
    suspend fun getAuthCode(request: AuthRequest) : CodeResponse {
        logger.info { "get auth code by request $request" }
        return authenticationService.creatSessionAndGetAuthCode(request);
    }

    @MessageMapping("get-tokens")
    suspend fun getTokensByAuthCode(request: AuthRequest) {
        logger.info { "get token by auth code $request" }
        //return AuthResponse(true)
    }
}
